src/main/resources/templates/checkout/application.html

```diff
@@ -0,0 +1,107 @@
+<!DOCTYPE html>
+<html lang="ja">
+<head>
+    <meta charset="UTF-8">
+    <title>貸出申請</title>
+</head>
+<body>
+<h1>貸出申請</h1>
+<input type="number" name="equipmentId" id="equipmentId">
+<button type="submit" id="addList">＋</button>
+<div id="message" th:if="${message}" th:text="${message}">message info</div>
+
+<div id="equipment-info">
+    <!-- ここに備品情報を表示 -->
+</div>
+
+<form th:action="@{/checkout/application}" method="post" >
+    <input type="hidden" name="equipmentIds" id="equipmentIds" value="" >
+    <button type="submit">申請</button>
+</form>
+</body>
+</html>
+
+<script>
+    // 備品のIDを格納する配列
+    let equipmentIds = [];
+
+    // Elementを取得
+    const equipmentIdElement = document.getElementById('equipmentId');
+    const equipmentInfoElement = document.getElementById('equipment-info');
+    const equipmentIdsElement = document.getElementById('equipmentIds');
+    const messageElement = document.getElementById('message');
+
+    // ＋ボタンをクリックしたときの処理
+    document.getElementById('addList').addEventListener('click', () => {
+        const equipmentIdValue = equipmentIdElement.value;
+
+        // equipmentIdsにequipmentIdの値が含まれていない場合の処理
+        if (!equipmentIds.includes(equipmentIdValue)) {
+            // 対象の備品情報を取得
+            getEquipmentInfo(equipmentIdValue);
+        } else {
+            messageElement.innerText = "すでに追加されています。";
+        }
+        equipmentIdElement.value = "";
+    });
+
+    // ✕ボタンをクリックしたときの処理
+    document.getElementById('equipment-info').addEventListener('click', (e) => {
+        if (e.target.tagName === 'BUTTON') {
+            // クリックされた要素の親要素のidを取得
+            const equipmentId = e.target.parentElement.id;
+
+            // equipmentIdsから対象のequipmentIdを削除
+            equipmentIds = equipmentIds.filter(id => id !== equipmentId);
+
+            // 削除後のequipmentIdsをhiddenに設定
+            equipmentIdsElement.value = equipmentIds;
+
+            // 要素を削除
+            e.target.parentElement.remove();
+
+            messageElement.innerText = "";
+        }
+
+        if(equipmentIds.length === 0) {
+            messageElement.innerText = "備品を選択してください。";
+        }
+    });
```

Comment:
早期リターン推奨です。
※早期リターンの影響を考慮してないため、いい感じに修正してください。

```suggestion
    document.getElementById('equipment-info').addEventListener('click', (e) => {
        if (e.target.tagName !== 'BUTTON') return;
        
         // リストから対象の備品を削除
         const equipmentId = e.target.parentElement.id;
         equipmentIds = equipmentIds.filter(id => id !== equipmentId);
         equipmentIdsElement.value = equipmentIds;
         e.target.parentElement.remove();
        
        // メッセージを更新
        if (equipmentIds.length === 0) messageElement.innerText = "備品を選択してください。";
        else  messageElement.innerText = "";
    });
```

Reply:
JSは一緒に編集したため解決とします。